src/views/Course.vue

```diff
@@ -931,62 +936,53 @@ const deleteNote = () => {
-  } else {
-    addCollaboratorError.value = null
-    collaborators.value.push(pendingCollaborator.value)
-    alertScreenReader(`${pendingCollaborator.value.firstName} ${pendingCollaborator.value.lastName} added as a collaborator.`)
+const addCollaboratorConfirm = (collaborator) => {
+  if (collaborator) {
+    const exists = collaborators.value.some(c => c.uid === collaborator.uid)
+    if (exists) {
+      addCollaboratorError.value = `${collaborator.firstName} ${collaborator.lastName} is already a collaborator.`
+      alertScreenReader(addCollaboratorError.value)
+    } else {
+      addCollaboratorError.value = null
+      collaborators.value.push(collaborator)
+      alertScreenReader(`${collaborator.firstName} ${collaborator.lastName} added as a collaborator.`)
+    }
+    putFocusNextTick('input-collaborator-lookup-autocomplete')
   }
-  pendingCollaborator.value = null
-  putFocusNextTick('input-collaborator-lookup-autocomplete')
 }
 
 const addCanvasSiteById = () => {
-  if (this.pendingCanvasSiteId.value && !this.isCanvasSiteIdStaged(this.pendingCanvasSiteId)) {
-    getCourseSite(this.pendingCanvasSiteId.value).then(data => {
+  if (pendingCanvasSiteId.value && !isCanvasSiteIdStaged(pendingCanvasSiteId)) {
+    getCourseSite(pendingCanvasSiteId.value).then(data => {
       if (data) {
-        this.publishCanvasSites.value.push(data)
+        publishCanvasSites.value.push(data)
         alertScreenReader(`${data.name} added.`)
         putFocusNextTick('input-canvas-site-id')
       }
     })
   }
-  this.pendingCanvasSiteId.value = null
+  pendingCanvasSiteId.value = null
 }
 
 const addCanvasSiteConfirm = () => {
-  if (this.pendingCanvasSite && !this.isCanvasSiteIdStaged(this.pendingCanvasSite.canvasSiteId)) {
-    this.publishCanvasSites.push(this.pendingCanvasSite)
+  if (pendingCanvasSite.value && !isCanvasSiteIdStaged(pendingCanvasSite.canvasSiteId)) {
+    publishCanvasSites.push(pendingCanvasSite)
   }
-  alertScreenReader(`${this.pendingCanvasSite.name} added.`)
+  alertScreenReader(`${pendingCanvasSite.name} added.`)
   putFocusNextTick('select-canvas-site')
-  this.pendingCanvasSite = null
-}
-
-const addCollaboratorPending = (collaborator) => {
-  if (collaborator) {
-    this.pendingCollaborator.value = collaborator
-    this.addCollaboratorError.value = null
-    this.$nextTick(() => {
-      this.alertScreenReader(`${this.pendingCollaborator.value.firstName} ${this.pendingCollaborator.value.lastName} selected.`)
-    })
-  }
+  pendingCanvasSite.value = null
 }
 
 const removeCanvasSite = (canvasSiteId, index) => {
-  const nextFocusIndex = (index + 1 === this.$_.size(this.publishCanvasSites)) ? index - 1 : index + 1
-  const nextFocusSiteId = get(this.publishCanvasSites, `${nextFocusIndex}.canvasSiteId`)
-  const canvasSite = find(this.publishCanvasSites, c => c.canvasSiteId === canvasSiteId)
+  const nextFocusIndex = (index + 1 === $_.size(publishCanvasSites)) ? index - 1 : index + 1
```

Comment:
@mohamalnadi the `$_.` still needs to be removed from each lodash function (size, find, first) but if this is a work in progress I don't mind merging it as-is.